onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -63,11 +63,22 @@ def check_and_normalize_provider_args(
     provider_name_to_options = collections.OrderedDict()
 
     def set_provider_options(name, options):
+        if (
+            os.environ.get("ORT_FALLBACK_CUDA_EP_TO_ROCM_EP", "0") == "1"
```

Comment:
We've avoided env variables so far for configs. Let's continue this convention.

Reply:
This is not config, this is a walkaround. This fallback logic happens right before the construction of Session, so entry `ORT_FALLBACK_CUDA_EP_TO_ROCM_EP` neither fits into session options nor provider options.